app/src/main/java/org/gem/indo/dooit/views/main/MainActivity.java

```diff
@@ -53,6 +66,54 @@
     @Inject
     Persisted persisted;
 
+    private InputStream resolveInputStream(Uri streamUri) throws IOException {
+        HttpURLConnection urlConnection = null;
+        try {
+
+            URL remote = new URL(streamUri.toString());
+            urlConnection = (HttpURLConnection) remote.openConnection();
+            urlConnection.setDoInput(true);
+            urlConnection.setRequestProperty("Accept", "application/json");
+            urlConnection.setRequestProperty("Authorization","Token " + persisted.getToken());
+            urlConnection.connect();
+            InputStream in = urlConnection.getInputStream();
+            return in;
+        }catch (Exception ioe){
+            Crashlytics.log("cannot download " + streamUri);
+
+        }
+        return getContentResolver().openInputStream(streamUri);
+
+    }
+    private class ProfileImageTask extends AsyncTask<Void,Integer,RoundedBitmapDrawable>{
```

Comment:
Why use AsyncTask this defies the purpose of using Fresco which handles Image Streaming and Scaling. This solution  downloads this image every time and does not use Fresco Image Cache ?